src/compiler/hxb/hxbData.ml

```diff
@@ -122,12 +122,23 @@ let error (s : string) =
 	Printf.eprintf "[error] %s\n" s;
 	raise (HxbFailure s)
 
-let hxb_version = 1
+(*
+	With the exception of 1.0 => 2.x because we introduced minor version in 2.1,
+	major versions are incompatible with each other, while minor version mismatch
+	will be handled by the reader (for retro compatibility only)
+
+	When further bumping hxb major:
+	- code related to the 1.0 => 2.0 exception can be dropped (`if major == 1 then 2 else major` in `HxbReader.read`)
+	- all minor checks in hxb reader become obsolete and can be dropped too
+*)
+let hxb_major = 2
+let hxb_minor = 1
```

Comment:
Shouldn't this start at 0?

Reply:
The hack for retro compatibility with preview1 is to consider preview1 to be `2.0` (instead of `1.0`) so that we're back to comparing minor versions

Reply:
But why do we check the minor version in the first place?

Reply:
To know when some data should (or not) be read, like https://github.com/HaxeFoundation/haxe/pull/12365/files#diff-e11692c26a1b5eb50b0ff3c69d1faa85d755afd3eb2e2daba537d844d8eddb21R1587-R1591 for 12351

Reply:
Yes but then why not check `hxb_major >= 2` for this one?

Reply:
It would work (well it wouldn't with current impl) too, but for the next changes we'll always want to check minor versions, never major (if the major version didn't match we wouldn't be in hxb reader in the first place).
I figured I'd use that same behavior from the beginning